build/build_suite_test.go

```diff
@@ -179,4 +181,170 @@ var _ = Describe("Test Build utilities:", func() {
 		})
 	})
 
+	Describe("Test NewEndpointSecretFromEnv", func() {
+		name := "name"
+		namespace := "namespace"
+		username := os.Getenv(manager.UsernameKey)
+		password := os.Getenv(manager.PasswordKey)
+		want := &v1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Type: v1.SecretTypeOpaque,
+			Data: map[string][]byte{
+				manager.UsernameKey: []byte(username),
+				manager.PasswordKey: []byte(password),
+			},
+			StringData: map[string]string{
+				manager.RegionNameKey:  os.Getenv(manager.RegionNameKey),
+				manager.DomainNameKey:  os.Getenv(manager.DomainNameKey),
+				manager.ProjectNameKey: os.Getenv(manager.ProjectNameKey),
+				manager.AuthUrlKey:     os.Getenv(manager.AuthUrlKey),
+				manager.InterfaceKey:   os.Getenv(manager.InterfaceKey),
+			},
+		}
+		keystoneRegion := os.Getenv(manager.KeystoneRegionNameKey)
+		if keystoneRegion != "" {
+			want.StringData[manager.KeystoneRegionNameKey] = keystoneRegion
+		}
+
+		secret, err := NewEndpointSecretFromEnv(name, namespace)
+		Expect(err).To(BeNil())
+		Expect(secret).To(Equal(want))
+	})
+
+	Describe("Test buildEndpointSecret", func() {
+		gClient := &gophercloud.ServiceClient{}
```

Comment:
missing "Context" & "It" intentionally?
i think its good to have from readability perspective.

Reply:
Updated